src/input.rs

```diff
@@ -141,10 +150,39 @@ pub trait Input: Sized {
     where
         O: Output,
     {
-        crate::run_result!(self)
+        let context = Context::production();
+        run_result_with_context(context, self)
     }
 }
 
+pub(crate) fn run_result_with_context<Stdout, Stderr, I, O>(
```

Comment:
This (and `run_result_with_context_unit`) seem like they should be methods on `Input`. But then they inherit the `pub` visibility from that trait. So I made them `pub(crate)` standalone functions.